include/mellon/impl/extensions.h

```diff
@@ -23,6 +23,7 @@ struct FUTURES_EMPTY_BASE future_type_based_extensions<expect::expected<T>, Fut,
             typename R = std::invoke_result_t<F, U&&>>
   auto then(F&& f) && noexcept {
     // TODO what if `F` returns an `expected<U>`. Do we want to flatten automagically?
+    // mpoeter - I think it would make sense!
```

Comment:
I dislike doing things automagically. We should add an extra function for that.

Reply:
Alright, then I'll remove my comment and the TODO.